agent/jvm/src/main/java/org/jolokia/jvmagent/CleanupThread.java

```diff
@@ -68,9 +68,10 @@ public void run() {
     /**
      * Stop the server.
      */
-    public void stopServer() {
+    public void stopServer() throws InterruptedException {
         active = false;
         interrupt();
+        join(); // once interrupted, wait for server to stop
```

Comment:
Thanks, makes sense for 1.x, too ;-)

I will backport this to Jolokia 1.x if you dont mind.